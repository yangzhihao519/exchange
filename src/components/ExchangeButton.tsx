import React from 'react'

interface ExchangeButtonProps {
    onClickExchange: () => void,
    disabled: boolean
}

class ExchangeButton extends React.Component<ExchangeButtonProps>{

    render() {
        let className = this.props.disabled? 'exchange-button primary-button disabled' : 'exchange-button primary-button'

        return (<div className={className} onClick={this.props.onClickExchange}> 
            Exchange
        </div>)
    }
    
}

export default ExchangeButton