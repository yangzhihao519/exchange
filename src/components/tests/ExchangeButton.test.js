import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ExchangeButton from '../ExchangeButton';

const mockFunction = jest.fn();

describe('ExchangeButton', () => {
  it('renders without crashing given the required props', () => {
    const props = {
        onClickExchange: mockFunction,
        disabled: false
    }
    const wrapper = shallow(<ExchangeButton {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})