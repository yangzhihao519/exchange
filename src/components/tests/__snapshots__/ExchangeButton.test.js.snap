// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeButton renders without crashing given the required props 1`] = `
<div
  className="exchange-button primary-button"
  onClick={[MockFunction]}
>
  Exchange
</div>
`;
